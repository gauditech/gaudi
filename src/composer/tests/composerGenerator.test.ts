import { compile, compose, parse } from "@src/index";

describe("generator composer", () => {
  it("succeeds for JS client generator", () => {
    const bp = `
      generate client {
        target js
        api entrypoint
      }
    `;

    const def = compose(compile(parse(bp)));
    const generator = def.generators[0];

    expect(generator).toMatchSnapshot();
  });

  it("succeeds for multiple client generators", () => {
    const bp = `
      generate client {
        target js
        api entrypoint
        // without output
      }

      generate client {
        target js
        api model
        output "a/b/c"
      }
    `;

    const def = compose(compile(parse(bp)));
    const generators = def.generators;

    expect(generators).toMatchSnapshot();
  });

  it("fails for multiple generators with the same target/api", () => {
    const bp = `
      generate client {
        target js
        api entrypoint
        output "a/b/c"
      }

      generate client {
        target js
        api entrypoint
        output "a/b/c"
      }
    `;

    expect(() => compose(compile(parse(bp)))).toThrowErrorMatchingInlineSnapshot(
      `"Found duplicate generator "generator-client", targeting the same target "js" and api "entrypoint""`
    );
  });
});
