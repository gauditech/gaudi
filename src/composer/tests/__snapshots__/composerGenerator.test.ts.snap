// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator composer succeeds for JS client generator 1`] = `
{
  "api": "entrypoint",
  "kind": "generator-client",
  "output": undefined,
  "target": "js",
}
`;

exports[`generator composer succeeds for multiple client generators 1`] = `
[
  {
    "api": "entrypoint",
    "kind": "generator-client",
    "output": undefined,
    "target": "js",
  },
  {
    "api": "model",
    "kind": "generator-client",
    "output": ""a/b/c"",
    "target": "js",
  },
]
`;
